### PR TITLE
Add ability to strip binaries in cmake

### PR DESCRIPTION
**Pull request by alesapin**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add an ability to build stripped binaries with cmake.

### CODE REVIEW

File: cmake/tools.cmake [hunk 1]
```diff
@@ -169,3 +169,31 @@ if (OBJCOPY_PATH)
 else ()
     message (FATAL_ERROR "Cannot find objcopy.")
 endif ()
+
+if (COMPILER_GCC)
```
Comment (alesapin) [inline cmake/tools.cmake:175]: copypaste

File: cmake/strip.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+BINARY_PATH=$1
+BINARY_NAME=$(basename $BINARY_PATH)
+DESTINATION_STRIPPED_DIR=$2
+OBJCOPY_PATH=${3:objcopy}
+READELF_PATH=${4:readelf}
+
+BUILD_ID=$($READELF_PATH -n $1 | sed -n '/Build ID/ { s/.*: //p; q; }')
```
Comment (alesapin) [inline cmake/strip.sh:9]: Now in some cases our build system depend on bash.

File: CMakeLists.txt [hunk 3]
```diff
@@ -263,6 +263,11 @@ if (OBJCOPY_PATH AND YANDEX_OFFICIAL_BUILD AND (NOT CMAKE_TOOLCHAIN_FILE))
     set (USE_BINARY_HASH 1)
 endif ()
 
+# Allows to build stripped binary in a separate directory
+if (OBJCOPY_PATH AND READELF_PATH)
```
Comment (Felixoid) [inline CMakeLists.txt:267]: Does it mean that `BUILD_STRIPPED_BINARIES_PREFIX` will be always set according to https://github.com/ClickHouse/ClickHouse/pull/35196/files#diff-20dc468ff5043ade1489346dbd70f621d828a1c8c6a439ef9b0afc3bc295a0a0R178 ?

I don't really think we always want strip it

Comment (Felixoid) [reply CMakeLists.txt:267]: Figured it out in slack

**Comment by Felixoid** [comment]
The fuzzer's issue is created, merge it now https://github.com/ClickHouse/ClickHouse/issues/35210

**Pull request merged**